### DISCUSSION TITLE
Option to repeat params for arrays instead of using commas

### DISCUSSION THREAD

**Discussion by lpbonomi** [47ng/nuqs#820] [Ideas] [answered]
This is how Next.js 15 parses the following URL generated by nuqs:
```
https://localhost:3000/deals?templateIds=6,2 
{ templateIds: '6,2' }
```

This is how Next.js 15 parses the following URL:
```
https://localhost:3000/deals?templateIds=6&templateIds=2 
{ templateIds: ['6', '2'] }
```

It could be useful to add this option to avoid manually parsing arrays in the backend

**Comment by franky47** [comment, marked as answer]
This is the topic of #484, and I agree it would be nice to support this standard.

For an immediate way of parsing them on the server, you can do the following:
```ts
import { parseAsArrayOf, parseAsInteger } from 'nuqs/server'

// Note: you can share this definition with useQueryState(s) on the client, the serializer helper etc..
const templateIdParser = parseAsArrayOf(parseAsInteger).withDefault([])

const templateIds = templateIdParser.parseServerSide('6,2')
//    ^? [6, 2]
```

Note that a more scalable way of "loading" search params from an object describing keys and associated parsers (as we use in many parts of the nuqs API) is coming in nuqs 2.3.0 in the form of loaders, see #818.
